packages/sheets-filter-ui/src/commands/commands/sheets-filter.command.ts

```diff
@@ -14,15 +14,90 @@
 import { CommandType, ICommandService, IUndoRedoService, IUniverInstanceService, LocaleService, Quantity, sequenceExecute, UniverInstanceType } from '@univerjs/core';
 import { MessageType } from '@univerjs/design';
 import type { ISheetCommandSharedParams } from '@univerjs/sheets';
-import { expandToContinuousRange, isSingleCellSelection, SheetsSelectionsService } from '@univerjs/sheets';
+import { expandToContinuousRange, getSheetCommandTarget, isSingleCellSelection, SheetsSelectionsService } from '@univerjs/sheets';
 import type { FilterColumn, IAutoFilter, IFilterColumn, IReCalcSheetsFilterMutationParams, ISetSheetsFilterCriteriaMutationParams, ISetSheetsFilterRangeMutationParams } from '@univerjs/sheets-filter';
 import { ReCalcSheetsFilterMutation, RemoveSheetsFilterMutation, SetSheetsFilterCriteriaMutation, SetSheetsFilterRangeMutation, SheetsFilterService } from '@univerjs/sheets-filter';
 import { IMessageService } from '@univerjs/ui';
 
+export interface ISetSheetFilterRangeCommandParams extends ISheetCommandSharedParams {
+    range: IRange;
+}
+
+export const SetSheetFilterRangeCommand: ICommand<ISetSheetFilterRangeCommandParams> = {
+    id: 'sheet.command.set-filter-range',
+    type: CommandType.COMMAND,
+    handler: (accessor, params: ISetSheetFilterRangeCommandParams) => {
+        const sheetsFilterService = accessor.get(SheetsFilterService);
+        const commandService = accessor.get(ICommandService);
+        const undoRedoService = accessor.get(IUndoRedoService);
+        const instanceSrv = accessor.get(IUniverInstanceService);
+
+        const { unitId, subUnitId, range } = params;
+
+        const commandTarget = getSheetCommandTarget(instanceSrv, params);
+        if (!commandTarget) return false;
+
+        const filterModel = sheetsFilterService.getFilterModel(unitId, subUnitId);
+        if (!filterModel) return false;
+
+        if (range.endRow === range.startRow) {
+            const messageService = accessor.get(IMessageService, Quantity.OPTIONAL);
+            const localeService = accessor.get(LocaleService);
+            messageService?.show({ type: MessageType.Warning, content: localeService.t('sheets-filter.command.not-valid-filter-range') });
+            return false;
+        }
+
+        // Execute the command to set filter range and prepare undo redo.
+        const redoMutation = { id: SetSheetsFilterRangeMutation.id, params: { unitId, subUnitId, range } };
+        const result = commandService.syncExecuteCommand(redoMutation.id, redoMutation.params);
+        if (result) {
+            undoRedoService.pushUndoRedo({
+                unitID: unitId,
+                undoMutations: [{ id: RemoveSheetsFilterMutation.id, params: { unitId, subUnitId } }],
+                redoMutations: [redoMutation],
+            });
+        }
+
+        return result;
+    },
+};
+
+export const RemoveSheetFilterCommand: ICommand<ISheetCommandSharedParams> = {
```

Comment:
here can use typedef , but it not must case

Reply:
Could you please give an example? @VicKun4937